src/aatoolbox/config/countryconfig.py

```diff
@@ -26,33 +26,73 @@ class CodABConfig(BaseModel):
     """
 
     hdx_dataset_name: str
-    layer_base_name: str  # TODO: validate that it has {admin_level}
+    layer_base_name: str
     admin_level_max: int
     custom_layer_names: Optional[list]
 
+    @validator("layer_base_name")
+    def _validate_layer_base_name(cls, layer_base_name):
+        if "{admin_level}" not in layer_base_name:
+            raise ValueError(
+                "The layer base name must contain "
+                "an {admin_level} placeholder."
+            )
+        return layer_base_name
+
 
 class CountryConfig(BaseModel):
     """Country configuration."""
 
     iso3: str
-    codab: CodABConfig
+    codab: Optional[CodABConfig]
+
+    @validator("iso3")
+    def _validate_iso3(cls, iso3):
+        if len(iso3) != 3 or not str.islower(iso3) or not str.isalpha(iso3):
+            raise ValueError("ISO3 must be a lowercase three letter string.")
+        return iso3
 
 
 def create_country_config(iso3: str) -> CountryConfig:
     """
-    Return a country configuration object.
+    Return a country configuration object from AA Toolbox.
 
     Parameters
     ----------
     iso3 : str
-        Country ISO3, must e exactly 3 characters long
+        Country ISO3, must be exactly 3 letters long
 
     Returns
     -------
     CountryConfig instance
     """
-    # TODO: validate iso3
-    parameters = parse_yaml(
-        Path(__file__).parent.resolve() / f"countries/{iso3}.yaml"
-    )
+    if len(iso3) != 3 or not str.isalpha(iso3):
+        raise ValueError("ISO3 must be a three letter string.")
+    try:
+        parameters = parse_yaml(
+            Path(__file__).parent.resolve() / f"countries/{iso3.lower()}.yaml"
+        )
+    except FileNotFoundError as err:
+        raise FileNotFoundError(
+            f"A configuration file for {iso3.upper()} is not yet available "
+            f"in AA Toolbox. Try using a custom configuration file with "
+            f"create_custom_country_config instead, or contact us to "
+            f"request that we add this country."
+        ) from err
     return CountryConfig(**parameters)
+
+
+def create_custom_country_config(filename: Union[str, Path]) -> CountryConfig:
```

Comment:
very small but maybe name it filepath instead of filename so it is more clear it should be the whole path?

Reply:
Good point, done!